### PR TITLE
refactor: core: Removing constructor params from DepositRecord

### PR DESCRIPTION
**Pull request by davidprepo**
Spec: https://www.notion.so/prepo/SC-Backlog-Tasks-0c6725bb870f4c22b7ba771347333662#2a3b3ac5ade246db99dafcdaf6bd96d3

### CODE REVIEW

**Comment by ramenforbreakfast** [comment]
Hey @davidprepo there is a test failing? Seems like a setting of deposit limits was forgotten in `DepositHook.test.ts`?

**Comment by ramenforbreakfast** [comment]
> Hey @davidprepo there is a test failing? Seems like a setting of deposit limits was forgotten in `DepositHook.test.ts`?

I have fixed this by replacing the `DepositRecord` used in `DepositHook.test.ts` with a fake.

File: apps/smart-contracts/core/test/WithdrawHook.test.ts [hunk 1]
```diff
@@ -38,10 +37,7 @@ describe('=> WithdrawHook', () => {
 
   beforeEach(async () => {
     ;[deployer, user, treasury] = await ethers.getSigners()
-    depositRecord = await smockDepositRecordFixture(
-      TEST_GLOBAL_DEPOSIT_CAP,
-      TEST_ACCOUNT_DEPOSIT_CAP
-    )
+    depositRecord = await smockDepositRecordFixture()
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/WithdrawHook.test.ts:40]: this should use a fake

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

**Pull request merged**